docs/usage/implementing-methods.rst

```diff
@@ -1,2 +1,286 @@
+    custom_method.infer(n_sim=4000) # {'n_batches': 4, 'n_sim': 4000, 'filtered_outputs': ...}
+
+    # Or use it iteratively
+    custom_method.set_objective(n_sim=6000)
+
+    custom_method.iterate()
+    assert custom_method.finished == False
+    # Investigate the current state
+    custom_method.extract_result()  # {'n_batches': 5, 'n_sim': 5000, 'filtered_outputs': ...}
+
+    self.iterate()
+    assert custom_method.finished
+    custom_method.extract_result()  # {'n_batches': 6, 'n_sim': 6000, 'filtered_outputs': ...}
+
+This works, because the state is stored into the ``custom_method`` instance, and we only
+change the objective. Also ELFI calls ``iterate`` internally in the ``infer`` method.
+
+The last finishing touch to our algorithm is to convert the ``state`` dict to a more user
+friendly format in the ``extract_result`` method. First we want to convert the list of
+filtered arrays from the batches to a numpy array. We will then wrap the result to a
+``elfi.methods.results.Sample`` object and return it instead of the ``state`` dict. Below
+is the final complete implementation of our inference method class::
+
+   import numpy as np
+
+   from elfi.methods.parameter_inference import ParameterInference
+   from elfi.methods.results import Sample
+
+
+   class CustomMethod(ParameterInference):
+       def __init__(self, model, discrepancy_name, threshold, **kwargs):
+           # Create a name list of nodes whose outputs we wish to receive
+           output_names = [discrepancy_name] + model.parameter_names
+           super(CustomMethod, self).__init__(model, output_names, **kwargs)
+
+           self.threshold = threshold
+           self.discrepancy_name = discrepancy_name
+
+           # Prepare lists to push the filtered outputs into
+           self.state['filtered_outputs'] = {name: [] for name in output_names}
+
+       def set_objective(self, n_sim):
+           self.objective['n_sim'] = n_sim
+
+       def update(self, batch, batch_index):
+           super(CustomMethod, self).update(batch, batch_index)
+
+           # Make a filter mask (logical numpy array) from the distance array
+           filter_mask = batch[self.discrepancy_name] <= self.threshold
+
+           # Append the filtered parameters to their lists
+           for name in self.output_names:
+               values = batch[name]
+               self.state['filtered_outputs'][name].append(values[filter_mask])
+
+       def extract_result(self):
+           filtered_outputs = self.state['filtered_outputs']
+           outputs = {name: np.concatenate(filtered_outputs[name]) for name in self.output_names}
+
+           return Sample(
+               method_name='CustomMethod',
+               outputs=outputs,
+               parameter_names=self.parameter_names,
+               discrepancy_name=self.discrepancy_name,
+               n_sim=self.state['n_sim'],
+               threshold=self.threshold
```

Comment:
Hmm, I don't really speak about the seed in this tutorial and it's included in the **kwargs.